In [2]:
#File Reading
#Emma Anderson, Big Data, 9/7/16
#open and save the file for reading
data_file = open("FreBridge1.csv", "r");
#create an empty list to store the data
data_list = [];
#put all lines from the file into a list
for ii in data_file:
    data_list.append(ii);

#remove the first element (column header)
del data_list[0];
#create empty lists for westbound and eastbound bikes
west = [];
east = [];    
for jj in range(len(data_list)):
    #make into a list of lists of just numbers
    data_list[jj] = data_list[jj].split(",");
    #add data to specific lists for west and east
    west.append(int(data_list[jj][1]));
    east.append(int(data_list[jj][2]));



In [3]:
import statistics 
import math


In [4]:
print("The mean of west is", statistics.mean(west))
print("The median of west is", statistics.median(west))

print("The mean of east is", statistics.mean(east))
print("The median of east is", statistics.median(east))



The mean of west is 1497.5666666666666
The median of west is 1573.5
The mean of east is 1540.4333333333334
The median of east is 1648.0



These numbers represent the number of bikers and pedestrians who go through the west and east entrance of the Fremont Bridge on a dialy basis. For both sets the mean and median are fairly close (respectively) which indicates that neither are significantly more accurate or telling of more information. Not only does a mode not exist for both the east and west entrance, but even if there was, it wouldn't be very useful with this data. There are about 30 data points for each set (west and east) with a range of ~500 to ~2500. We could calculate the maximum and minimun number of people who used the bridge in a single day to see how close those numbers are to the data mean and median in order to determine whether the median or mean is more accurate. Suppose the maxium or minimun was extreme and skewing the data, then the median would most likely be more telling than the mean. On average, Wednesday is the most popular day for pedestrians and bikers to pass through the west and east entrance. This was calculated with the given data, an average was drawn for each day of the week and from there the maximum average was Wednesday. Bikers follow the flow of traffic however pedestrians can walk on either side of the bridge in either direction. The data for east and west are not significantly different on a day to day basis so we can look at them as a whole. 



In [5]:
tues= []
weds=[]
thurs=[]
fri=[]
sat=[]
sun=[]
mon=[]

for i in range(0, 28, 7): 
    tues.append (west[i])
    weds.append (west[i+1])
    thurs.append (west[i+2])
    fri.append (west[i+3])
    sat.append (west[i+4])
    sun.append (west[i+5])
    mon.append (west[i+6])

print("DATA FOR WEST ENTRANCE")
print ("The mean for Tuesday is", statistics.mean (tues))
print ("The mean for Wednesday is", statistics.mean (weds))
print ("The mean for Thursday is", statistics.mean (thurs))
print ("The mean for Friday is", statistics.mean (fri))
print ("The mean for Saturday is", statistics.mean (sat))
print ("The mean for Sunday is", statistics.mean (sun))
print ("The mean for Monday is", statistics.mean (mon))




DATA FOR WEST ENTRANCE
The mean for Tuesday is 1777.75
The mean for Wednesday is 1966.75
The mean for Thursday is 1684.75
The mean for Friday is 1515.75
The mean for Saturday is 971
The mean for Sunday is 680.25
The mean for Monday is 1613.25


The day of the week in which the highest amount of crossing occured on the West entrance was Wednesday. 

In [6]:
tues= []
weds=[]
thurs=[]
fri=[]
sat=[]
sun=[]
mon=[]

for i in range(0, 28, 7): 
    tues.append (east[i])
    weds.append (east[i+1])
    thurs.append (east[i+2])
    fri.append (east[i+3])
    sat.append (east[i+4])
    sun.append (east[i+5])
    mon.append (east[i+6])

print("DATA FOR EAST ENTRANCE")
print ("The mean for Tuesday is", statistics.mean (tues))
print ("The mean for Wednesday is", statistics.mean (weds))
print ("The mean for Thursday is", statistics.mean (thurs))
print ("The mean for Friday is", statistics.mean (fri))
print ("The mean for Saturday is", statistics.mean (sat))
print ("The mean for Sunday is", statistics.mean (sun))
print ("The mean for Monday is", statistics.mean (mon))

DATA FOR EAST ENTRANCE
The mean for Tuesday is 1909.5
The mean for Wednesday is 2087.75
The mean for Thursday is 1806
The mean for Friday is 1579.75
The mean for Saturday is 876.25
The mean for Sunday is 592.75
The mean for Monday is 1625.75


The day of the week in which the highest amount of crossing occured on the East entrance was Wednesday. We left out a Tuesday and a Wednesday. 

In [7]:
data_file = open("FreBridge2.csv", "r");
data_list2 = [];
#put all lines from the file into a list
for tt in data_file:
    data_list2.append(tt);

#remove the first element (column header)
del data_list2[0];
#create empty lists for westbound and eastbound bikes
west = [];
east = [];    
for tt in range(len(data_list2)):
    #make into a list of lists of just numbers
    data_list2[tt] = data_list2[tt].split(",");
    #add data to specific lists for west and east
    west.append(int(data_list2[tt][1]));
    east.append(int(data_list2[tt][2]));

    
print("WEST entrance: ")
print("The mean of west is", statistics.mean(west))
print("The median of west is", statistics.median(west))
print("The mode of west is", statistics.mode(west))
print("EAST entrance:")
print("The mean of east is", statistics.mean(east))
print("The median of east is", statistics.median(east))
print("There is no mode for east")
print("Means for Fridays:")

fri=[]
for i in range(0, 28, 7): 
    fri.append (east[i+2])
print ("East=", statistics.mean (fri))


fri=[]
for i in range(0, 28, 7): 
    fri.append (west[i+2])
print ("West=", statistics.mean (fri))









WEST entrance: 
The mean of west is 1485.333799860042
The median of west is 1336
The mode of west is 1180
EAST entrance:
The mean of east is 1299.0881735479356
The median of east is 1183
There is no mode for east
Means for Fridays:
East= 1156.25
West= 1146


In [8]:
#finding the fridays in falls
fridayeast=[]
for o in range(2, 29, 7): 
    fridayeast.append (east[o])  
print(fridayeast)  
for o in range(337, 394, 7): 
    fridayeast.append (east[o])  
print(fridayeast)  
for o in range(695,758 , 7): 
    fridayeast.append (east[o])  
print(fridayeast)  
for o in range(1066,1122 , 7): 
    fridayeast.append (east[o])  
print(fridayeast)  
print(statistics.mean(fridayeast))


[1590, 989, 1012, 1034]
[1590, 989, 1012, 1034, 1070, 1975, 1870, 1739, 1581, 1513, 1529, 1391, 1266]
[1590, 989, 1012, 1034, 1070, 1975, 1870, 1739, 1581, 1513, 1529, 1391, 1266, 1642, 2065, 1917, 1841, 1175, 1491, 1215, 973, 1570]
[1590, 989, 1012, 1034, 1070, 1975, 1870, 1739, 1581, 1513, 1529, 1391, 1266, 1642, 2065, 1917, 1841, 1175, 1491, 1215, 973, 1570, 1512, 1898, 1539, 1370, 1540, 1616, 1709, 1590]
1507.4


The mean for Fridays (September-October) for the East entrance is: 1507.4

In [9]:
#finding the fridays in falls
fridaywest=[]
for o in range(2, 29, 7): 
    fridaywest.append (west[o])  
print(fridaywest)  
for o in range(337, 394, 7): 
    fridaywest.append (west[o])  
print(fridaywest)  
for o in range(695,758 , 7): 
    fridaywest.append (west[o])  
print(fridaywest)  
for o in range(1066,1122 , 7): 
    fridaywest.append (west[o])  
print(fridaywest)  
print(statistics.mean(fridaywest))



[1558, 1022, 965, 1039]
[1558, 1022, 965, 1039, 1178, 2070, 2248, 1874, 1670, 1659, 1722, 1624, 1307]
[1558, 1022, 965, 1039, 1178, 2070, 2248, 1874, 1670, 1659, 1722, 1624, 1307, 1546, 2122, 1986, 1893, 1336, 2218, 1742, 1236, 1208]
[1558, 1022, 965, 1039, 1178, 2070, 2248, 1874, 1670, 1659, 1722, 1624, 1307, 1546, 2122, 1986, 1893, 1336, 2218, 1742, 1236, 1208, 1425, 1863, 1461, 1314, 1477, 1503, 1756, 1497]
1583.9666666666667


The mean for Fridays (September-October) for the West entrance is: 1583.966

In [11]:

friday=[]
for m in range (len(fridayeast)): 
    friday.append(fridayeast[m] +fridaywest[m])

print("The mean for Fridays in the fall is: ")
print(statistics.mean(friday))
mean= statistics.mean(friday)
   
stdev = statistics.stdev(friday)
print("Standard Deviation for Fridays in the Fall")
print(statistics.stdev(friday))

n= len(friday)
print("Two standard deviation range:")
print( mean - (2)*(stdev/math.sqrt(n)))
print( mean + (2)*(stdev/math.sqrt(n)))


n= len(friday)
print("One standard deviation range:")
print( mean - (1)*(stdev/math.sqrt(n)))
print( mean + (1)*(stdev/math.sqrt(n)))



The mean for Fridays in the fall is: 
3091.366666666667
Standard Deviation for Fridays in the Fall
627.3945293053637
Two standard deviation range:
2862.2745758960777
3320.458757437256
One standard deviation range:
2976.820621281372
3205.9127120519615


I am 95% confident that the number of people who cross the Fremont Bridge on Friday, September 30th will be somewhere between 2862 and 3320 people. 

I am 67.5% confident that the number of people who cross the Fremont Bridge on Friday, September 30th will be somewhere between 2977 and 3206 people.

The basis for these claims is that the mean of Friday falls is 3091 and the ranges are one and two standard deviations out respectively. 